extension/src/popup/components/hardwareConnect/LedgerConnect/index.tsx

```diff
@@ -37,37 +47,68 @@ const parseLedgerError = (err: any): string => {
   return LEDGER_ERROR.OTHER;
 };
 
-export const LedgerConnect = ({ goBack }: { goBack?: () => void }) => {
-  const dispatch = useDispatch();
+export const LedgerConnect = () => {
+  const dispatch: AppDispatch = useDispatch();
   const [isConnecting, setIsConnecting] = useState(false);
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const {
+    hardwareWalletData: { transactionXDR },
+  } = useSelector(transactionSubmissionSelector);
+  const isSigningTransaction = !!transactionXDR;
 
-  // TODO - move to redux with bipPath
-  const [publicKey, setPublicKey] = useState("");
-  const isConnected = !!publicKey;
+  const [isConnected, setIsConnected] = useState(false);
   const [connectError, setConnectError] = useState("");
 
+  const closeOverlay = () => dispatch(closeHwOverlay());
+
   const handleConnect = async () => {
     setIsConnecting(true);
+    setConnectError("");
     try {
       const transport = await TransportWebUSB.request();
       const ledgerApi = new LedgerApi(transport);
       const response = await ledgerApi.getPublicKey(defaultStellarBipPath);
+      setIsConnected(true);
```

Comment:
Just spitballing an idea here: It's a bit confusing to have the app status be able to be both `isConnecting === true` and `isConnected === true` at the same time.

Would it maybe be more clear if there was a `connectionStatus` that could have the value of an enum?

for ex: `connectionStatus === CONNECTION_STATUS.CONNECTING` or `connectionStatus === CONNECTION_STATUS.CONNECTED`

It might make some of the UI logic below more messy, so if that's the case, I think you can ignore. But it'd be nice if these 2 var's were related

Reply:
ya  that's a good point 🤔 , maybe a good solution would moving the isConnected status to redux. Or else not ya  I agree an enum would also be more clear.

I'll explore that in my next PR to make it more clear